remotes/docker/config/hosts.go

```diff
@@ -122,8 +122,13 @@ func ConfigureHosts(ctx context.Context, options HostOptions) docker.RegistryHos
 			hosts[len(hosts)-1].capabilities = docker.HostCapabilityPull | docker.HostCapabilityResolve | docker.HostCapabilityPush
 		}
 
+		// explicitTLS indicates that TLS was explicitly configured and HTTP endpoints should
+		// attempt to use the TLS configuration before falling back to HTTP
+		var explicitTLS bool
+
 		var defaultTLSConfig *tls.Config
 		if options.DefaultTLS != nil {
+			explicitTLS = true
```

Comment:
Probably this shouldn't be true when DefaultScheme is "http"

Reply:
Whether the InsecureSkipVerify field should be checked?
eg: 
```go
if options.DefaultTLS != nil {
	if !options.DefaultTLS.InsecureSkipVerify {
	      explicitTLS = true
	}
	defaultTLSConfig = options.DefaultTLS
}
```

Reply:
Is the `DefaultScheme` only "http" there when explicitly set by the client? The "http" fallback should always be safe, but the tests might not expect that extra connection attempt. If the behavior is explicit from the client, we can keep as false, otherwise maybe we should see if the test could ignore the tls connection attempt.